Toolkit/ArcGISToolkit/TimeSlider.swift

```diff
@@ -0,0 +1,2539 @@
+            let endTimeLabelY = min(thumbEndTimeLabelY, tickLayerEndTimeLabelY)
+            currentExtentEndTimeLabel.frame = CGRect(x: endTimeLabelX, y: endTimeLabelY, width: endTimeLabelSize.width, height: endTimeLabelSize.height)
+        }
+        else {
+            currentExtentEndTimeLabel.string = ""
+            currentExtentEndTimeLabel.isHidden = true
+        }
+        
+        // Commit the transaction
+        CATransaction.commit()
+    }
+    
+    private func positionTickMarks() {
+        //
+        // Bail out if time steps are not available
+        guard let timeSteps = timeSteps, timeSteps.count > 0, isSliderVisible else {
+            return
+        }
+        
+        // Create tick marks based on
+        // time steps and it's calculated
+        // position on the slider
+        var tms = [TickMark]()
+        for i in 0..<timeSteps.count  {
+            let timeStep = timeSteps[i]
+            let tickX = CGFloat(position(for: timeStep.timeIntervalSince1970))
+            tms.append(TickMark(originX: tickX, value: timeStep))
+        }
+        
+        // If label mode is ticks then we need to
+        // find out the major/minor ticks and
+        // add labels for major ticks
+        if labelMode == .ticks {
+            var majorTickInterval = 2
+            var doMajorTickLabelsCollide = false
+            var firstMajorTickIndex = 0
+            let tickCount = tms.count
+            
+            // Calculate the largest number of ticks to allow between major ticks. This prevents scenarios where
+            // there are two major ticks placed undesirably close to the end.
+            let maxMajorTickInterval = Int(ceil(Double(tickCount / 2)))
+            
+            // Calculate the number of ticks between each major tick and the index of the first major tick
+            for i in majorTickInterval..<maxMajorTickInterval  {
+                let prospectiveInterval = i
+                var allowsEqualNumberOfTicksOnEnds = false
+                
+                // Check that the prospective interval between major ticks results in
+                // an equal number of minor ticks on both ends.
+                for m in stride(from: prospectiveInterval, to: tickCount, by: prospectiveInterval) {
+                    let totalNumberOfTicksOnEnds = tickCount - m + 1
+                    
+                    // If the total number of minor ticks on both ends (i.e. before and after the
+                    // first and last major ticks) is less than the major tick interval being tested, then we've
+                    // found the number of minor ticks that would be on the ends if we use this major tick interval.
+                    // If that total is divisible by two, then the major tick interval under test allows for an
+                    // equal number of minor ticks on the ends.

```

Comment:
Is it possible for there to be only one tick? If so, this would result in a runtime error as well.

One alternative would be to drop the first and last indexes:

```
for i in indices.dropFirst().dropLast() {
   …
}
```

Reply:
See above comment.